itests/markets_v1_deal_test.go

```diff
@@ -18,6 +18,7 @@ func TestMarketsV1Deal(t *testing.T) {
 	kit.QuietMiningLogs()
 	framework.SetLogLevel()
 	f := framework.NewTestFramework(ctx, t)
+	f.EnableLegacy = true
```

Comment:
I'd recommend passing the EnableLegacy flag to NewTestFramework as an option. We could add a TestFrameworkOptions struct with a field EnableLegacyDeals, similar to BitswapServerOptions (cmd/booster-bitswap/server.go)